andela/ah-kgl-avengers-backend
# #165020205 Creates Social login features with facebook and google

## What does this PR do?
Creates a social login with facebook and google using passport library

## Description of Task to be completed?
Currently, a new user cannot log in with their pre-existing social media accounts. He/She can only register and or login with email password combinations

## How should this be manually tested?
- clone the [repo](https://github.com/andela/ah-kgl-avengers-backend) cd into the project, check out into `ft-social-login-16520205` and do `npm install` to install the dependencies and `npm run dev` to start the server and `npm run test` to run the tests.
  - Create an app on [Facebook](https://developers.facebook.com) and [Google](https://console.developers.google.com)
   - Generate a token from either facebook or google then place the token in postman as JSON object like this : ```` "aacess_token": "token". ````

## Any background context you want to provide?
- This only uses facebook and google. Twitter is still taking it's time to confirm the app creation process.
   - Keep in mind google [OAuth v2](https://developers.google.com/oauthplayground) when generating the Google token.

## What are the relevant pivotal tracker stories?
[165020205](https://www.pivotaltracker.com/story/show/165020205)

## Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/33352484/55928604-a6917880-5c19-11e9-9b28-dc9a07154d57.png)

## Questions:
- Are we using mocking for testing or hitting the database straight
